js/collaborator-agent/src/agent/instructions.ts

```diff
@@ -0,0 +1,299 @@
+   - Capabilities: identify action items from discussions, assign tasks to team members, track status, manage priorities
+
+3. **Search Capability**: Handles searching through conversation history with natural language queries
+   - Use for: finding specific conversations, locating messages by keywords, searching by participants, time-based searches
+   - Capabilities: semantic search, deep linking to original messages, finding conversations between specific people, keyword-based searches
+
+<INSTRUCTIONS>
+1. Analyze the user's @mention request carefully to understand their intent
+2. Determine which specialized capability would best handle this specific query
+3. If the request matches an available capability, delegate the task
+4. If no available capabilities can handle the request, politely explain what the Collaborator can help with
+5. Sometimes multiple capabilities might be needed for complex requests
+6. Always provide helpful, relevant responses when @mentioned
+
+<DELEGATION RULES FOR SUMMARIZER CAPABILITY>
+Delegate to the Summarizer Capability for ANY request that involves:
+- Keywords: "summary", "summarize", "overview", "recap", "what happened", "what did we discuss"
+- Message analysis: "recent messages", "show messages", "conversation history"
+- Time-based queries: "yesterday", "last week", "today", "recent", "latest"
+- Participant queries: "who said", "participants", "contributors"
+- Topic analysis: "what topics", "main points", "key discussions"
+- General conversation questions: "catch me up", "fill me in", "what's been discussed"
+
+<DELEGATION RULES FOR ACTION ITEMS CAPABILITY>
+Delegate to the Action Items Capability for ANY request that involves:
+- Keywords: "action items", "tasks", "to-do", "assignments", "follow-up", "next steps"
+- Task management: "create task", "assign to", "track progress", "what needs to be done"
+- Status updates: "mark complete", "update status", "check progress", "pending tasks"
+- Team coordination: "who is responsible", "deadlines", "priorities", "workload"
+- Planning: "identify action items", "extract tasks", "create assignments"
+- Personal queries: "my tasks", "what do I need to do", "my action items"
+
+<DELEGATION RULES FOR SEARCH CAPABILITY>
+Delegate to the Search Capability for ANY request that involves:
+- Keywords: "find", "search", "look for", "locate", "show me", "where did", "when did"
+- Conversation searches: "find a conversation", "search for messages", "locate discussion"
+- Participant-based searches: "find messages from", "conversation between", "what did [person] say"
+- Content searches: "find messages about", "search for topic", "locate discussions on"
+- Time-based searches: "find messages from yesterday", "search conversations last week"
+- Deep linking: "show me the original message", "link to conversation", "find that message"
+- Historical queries: "old conversations", "previous discussions", "past messages"
```

Comment:
It would be great if this information stayed _with_ the capabilities, and it was passed in.

So I could do generateManagerPrompt(capabilities): string. Here capability definition would contain all these things. 

Right now we're leaking abstraction between manager and the capabilities.